_posts/2024-06-20-vector-radial-search.md

```diff
@@ -0,0 +1,360 @@
+        "knn": {
+            "my_vector": {
+                "vector": [7.1, 8.3],
+                "min_score": 0.95,
+                "filter": {
+                    "range": {
+                        "price": {
+                            "gte": 1,
+                            "lte": 5
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+```
+
+All documents with a score of 0.9 or higher and whose price is within the 1--5 range are returned in the results:
+
+```json
+{
+    ...
+    "hits": {
+        "total": {
+            "value": 1,
+            "relation": "eq"
+        },
+        "max_score": 0.98039204,
+        "hits": [
+            {
+                "_index": "knn-index-test",
+                "_id": "1",
+                "_score": 0.98039204,
+                "_source": {
+                    "my_vector": [7.0, 8.2],
+                    "price": 4.4
+                }
+            }
+        ]
+    }
+}
+```
+
+## Neural search with vector radial search
+
+Neural search also supports radial search. Here are some radial search neural query examples.
+
+The following example shows a search with a `k` value of `100` and a filter that includes a range query and a term query:
+
+```json
+GET /my-nlp-index/_search
+{
+  "query": {
+    "neural": {
+      "passage_embedding": {
+        "query_text": "Hi world",
+        "query_image": "iVBORw0KGgoAAAAN...",
+        "k": 100,
+        "filter": {
+          "bool": {
+            "must": [
+              {
+                "range": {
+                  "rating": {
+                    "gte": 8,
+                    "lte": 10
+                  }
+                }
+              },
+              {
+                "term": {
+                  "parking": "true"
+                }
+              }
+            ]
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+The following example shows a k-NN radial search with `min_score` of `0.95` and the same filter as the preceding query:
+
+```json
+GET /my-nlp-index/_search
+{
+  "query": {
+    "neural": {
+      "passage_embedding": {
+        "query_text": "Hi world",
+        "query_image": "iVBORw0KGgoAAAAN...",
+        "min_score": 0.95,
+        "filter": {
+          "bool": {
+            "must": [
+              {
+                "range": {
+                  "rating": {
+                    "gte": 8,
+                    "lte": 10
+                  }
+                }
+              },
+              {
+                "term": {
+                  "parking": "true"
+                }
+              }
+            ]
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+## Benchmarks
+
+We ran radial search benchmarks using the following cluster configuration.
+
+|OpenSearch version	|Leader nodes	|Leader node type	|Leader node disk space	|Data nodes	|Data node type	|Data node disk space	|Primary shard count	|Replica count	|Availability zone	|Test client type	|
```

Comment:
If "Availability zone" refers to an AWS Availability Zone, then "Zone" should be capitalized.